core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1548,6 +1552,25 @@ public void sendCommand(String command) {
         this.sendCommandPacket(command);
     }
 
+    @Override
+    public @NonNull String clientConnectionAddress() {
+        if (clientData == null) {
+            return "";
+        }
+        return clientData.getServerAddress();
+    }
+
+    @Override
+    public void sendSkin(@NonNull UUID player, @NonNull SkinData skinData) {
+        Objects.requireNonNull(player, "player uuid must not be null!");
+        Objects.requireNonNull(skinData, "skinData must not be null!");
+
+        PlayerEntity entity = this.entityCache.getPlayerEntity(player);
+        Objects.requireNonNull(entity, "player not found!");
```

Comment:
I'm not really sure if this is intuitive, especially because this part is not documented. I'd probably just make sendSkin a method that returns a bool. That bool says whether the actual player was found.

Reply:
will now return silently :)